models/forms/data_record.sql

```diff
@@ -22,6 +22,7 @@ SELECT
   document_metadata.saved_timestamp,
   to_timestamp((NULLIF(doc->>'reported_date'::text, ''::text)::bigint / 1000)::double precision) AS reported,
   doc->>'form' as form,
+  doc#>>'{fields,inputs,source}' as source,
```

Comment:
I don't see any benefits of breaking this into the separate columns. Moreover, that way is less performant and I've opened an [issue](https://github.com/medic/cht-pipeline/issues/186) to make this changes in every other places.

Reply:
Your suggestion to use direct access with` #>>` is an excellent approach for improving performance. If downstream processes don't need to select specific fields, inputs, or source attributes, separating them may not be necessary. 
However, if more granularity is required, we could balance performance and flexibility by separating the columns while using direct access instead of JSON object access. 
Please let me know what you think.